CHANGES.md

```diff
@@ -1,6 +1,420 @@
+* extended 'mason' to support three different usage modes  
+  (see https://chapel-lang.org/docs/1.28/mason-packages/start/whymason.html#the-three-modes-of-mason)
+* added support for adding git repositories as 'mason' dependencies  
+  (see https://chapel-lang.org/docs/1.28/mason-packages/guide/gitdependencies.html)
+* added a `mason modules` command to generate command-line flags
+* simplified initialization of 'mason' packages
+* added a check to ensure a 'mason' package exists before adding it
+* improved `mason test` errors to display relative, rather than absolute, paths
+
+Performance Optimizations / Improvements
+----------------------------------------
+* improved the performance of scan operations on `Block`-distributed arrays
+
+Platform-specific Performance Optimizations / Improvements
+----------------------------------------------------------
+
+Compilation-Time / Generated Code Improvements
+----------------------------------------------
+* improved average compilation time by using LLVM data structures
+
+Memory Improvements
+-------------------
+* improved memory requirements of 'IO' channel buffers for large reads/writes
+* addressed a memory leak in the 'TOML' module by switching to `shared` objects
+
+Documentation
+-------------
+* reworked the documentation for 'mason' to simplify navigation and clarify it  
+  (see https://chapel-lang.org/docs/1.28/mason-packages/index.html)
+* updated the 'Chapel Prerequisites' document to show uses of `LLVM_VERSION`  
+  (see https://chapel-lang.org/docs/1.28/usingchapel/prereqs.html)
+* refreshed the `range` API documentation in the language specification  
+  (see https://chapel-lang.org/docs/1.28/language/spec/ranges.html#range-type-queries)
+* refreshed the `locale` documentation to reflect stabilization improvements  
+  (see https://chapel-lang.org/docs/1.28/language/spec/locales.html#locale-methods)
+* clarified the behavior of generic formals with `out` intent in the spec  
+  (see https://chapel-lang.org/docs/1.28/language/spec/procedures.html#the-out-intent)
+* fixed a bug in the 'ArgumentParser' documentation where a flag was missing  
+  (see https://chapel-lang.org/docs/1.28/modules/packages/ArgumentParser.html#quickstart-example)
+
+Syntax Highlighting
+-------------------
+
+Example Codes
+-------------
+
+Build System Improvements
+-------------------------
+
+Portability / Platform-specific Improvements
+--------------------------------------------
+* generally improved Chapel portability to M1/Arm-based Macs
+* improved `chplvis` such that it can run on an M1 Mac
+* `chpl` now links dynamically with system LLVM and Clang libraries on Mac OS X
+* improved portability to configurations using GCC 12
+* worked around an ICC bug resulting in "unknown attribute" warnings
+* improved detection of system-installed Clang for Amazon Linux 2022
+* removed remaining specialized support for SunOS platforms
+
+GPU Computing
```

Comment:
Hi Brad,

Here are some URLs you can use for the TODO items you listed under "GPU Computing":

* https://chapel-lang.org/docs/1.28/modules/standard/GPU.html
* https://chapel-lang.org/docs/1.28/technotes/gpu.html#diagnostics-and-utilities
* https://chapel-lang.org/docs/1.28/modules/standard/GPU.html#GPU.gpuWrite
* https://chapel-lang.org/docs/1.28/modules/standard/GPU.html#GPU.assertOnGpu

And this is how I would suggest associating them with these items (arguably you could add the more general tech report link to the bottom two items as well but it feels redundant to me).

```
* added a new 'GPU' module for GPU-related utility and debugging features  
  (see https://chapel-lang.org/docs/1.28/technotes/gpu.html#diagnostics-and-utilities
   and https://chapel-lang.org/docs/1.28/modules/standard/GPU.html)
* added a `gpuWrite()` routine for console output from a GPU kernel  
  (https://chapel-lang.org/docs/1.28/modules/standard/GPU.html#GPU.gpuWrite)
* added an `assertOnGpu()` routine to ensure code is running on GPU as expected  
  (https://chapel-lang.org/docs/1.28/modules/standard/GPU.html#GPU.assertOnGpu)
```

Reply:
Thanks!